src/link_getter.py

```diff
@@ -0,0 +1,72 @@
+from bs4 import BeautifulSoup
+import requests
+import pandas as pd
+import time
+import json
+
+class link_getter :
+    def __init__(self, excel_name = "risk_dictionary.xlsx", base_url = "https://search.naver.com/search.naver?where=news&query="):
+        self.data_src = pd.read_excel(excel_name, sheet_name="1. 반도체 공급망 RISK 키워드 POOL")
+        self.base_url = base_url
+
+        self.data_col_list = ['주요 업체', '주요 원자재', '파운드리 기업', '반도체 기업', '기타 관련 기업', '반도체 소재 업체', '반도체 웨이퍼 업체', '반도체 장비 업체', '기구 및 협회', '반도체 공통용어', '반도체 기술 용어', '반도체 생태계 용어', '반도체 공정 용어', '반도체 소재']
+
+    # flag = 0 : Use all keywords from excel files. You can designate columns, if you don't want all columns.
+    # flag = 1 : Use your own data set. Use it if you want one or two words instead of whole things.
+    def get_link_by_naver(self, file_name, flag = 0, data = [], columns = [], repeat = 5):
+        _json = []
+        total_cnt = 0
+        
+        data = {
+            0 : self._get_data_by_col(columns),
+            1 : data
+        }.get(flag)
+
+        for key in data:
+            print(f"{key} started") #print log. Erase it if you don't want any log
+
+            for page_num in range(repeat):
+                url = self.base_url + key + self._add_param(page_num, '&sort=1')
+                html = requests.get(url)
+                soup = BeautifulSoup(html.text, 'html.parser')
+                print(url) #print log. Erase it if you don't want any log
+            
+                for elem in soup.select('#main_pack > section > div > div.group_news > ul > li > div > div > div.news_info > div.info_group > a.info'):
```

Comment:
This string is a hard coding. It should be made to ease to fix if changed.

Reply:
Solved. Now you can choose using link_getter.link_selectors